Open Judge System/Web/OJS.Web/Controllers/TempController.cs

```diff
@@ -302,5 +304,42 @@ public ActionResult OptimizeMysqlProblemSkeletonAndTestInputQueries()
                 $"Updated {changedSkeletonsCount} solution skeletons. <br/>" +
                 $"Updated {changedTestsCount} test inputs.");
         }
+        
+        public ActionResult MigrateSolutionSkeletons()
+        {
+            var updatedProblemsCount = 0;
+            this.problemsDataService.GetAll()
+                .Where(x => x.SolutionSkeleton != null && x.SubmissionTypes.Any() && !x
+                .ProblemSubmissionTypesSkeletons.Any())
+                .ToList()
+                .ForEach(
+                    p =>
+                    {
+                        try
+                        {
+                            updatedProblemsCount++;
```

Comment:
maybe move this after update so you don't have to decrease it in the catch block?